src/app/Dashboard/components/DashboardContainer.js

```diff
@@ -45,23 +48,43 @@ export class DashboardContainer extends Component {
     pointsEarned: PropTypes.number,
     activitiesLogged: PropTypes.number,
     fetchUserActivites: PropTypes.func,
+    loadCategories: PropTypes.func,
     userActivities: PropTypes.arrayOf(PropTypes.shape({})),
   };
 
   componentDidMount() {
-    const { fetchUserActivites } = this.props;
+    const { fetchUserActivites, loadCategories } = this.props;
     const token = getToken();
     const userInfo = getUserInfo(token);
     this.setState({ user: userInfo });
     fetchUserActivites(userInfo.id);
+    loadCategories();
+  }
+
+  logPointsModal = () => {
+    const { logPoints } = this.state;
+    this.setState({
+      logPoints: !logPoints,
+    });
   }
 
   render() {
     const {
       error, loading, pointsEarned, activitiesLogged, userActivities, society,
     } = this.props;
-    const { user } = this.state;
+    const {
+      user, logPoints,
+    } = this.state;
     let dashboardHtml;
+    let logPointsComponent;
+    if (logPoints) {
+      logPointsComponent = (
+        <LogPointsComponent
+          open={logPoints}
+          close={this.logPointsModal}
+        />
+      );
+    }
     if (loading) {
       dashboardHtml = <p>Loading ...</p>;
```

Comment:
Yes, we should. Having the text `Loading` alone does create a good experience for the user. But since it was not part of this scope I can create a story for this on PT and handle it separately.

Reply:
I have created a Loading component @engineer-kndegwa . It is in  this [PR](https://github.com/andela/andela-societies-frontend/pull/198)